File: Python/cadastroAlunos.py
alunos = []

print("Cadastre os alunos no sistema")
print("")

while True:

    nome = input("Nome do aluno: ")
    nota = float(input("Nota do aluno: "))

    alunos.append({"nome": nome, "nota": nota})

    continuar = input("Cadastrar outro aluno? (s/n): ")

    if continuar.lower() != "s":
        break

print("\nLista de alunos:")

soma = 0
maior_nota = 0
melhor_aluno = ""

for aluno in alunos:

    print(aluno["nome"], "-", aluno["nota"])

    soma += aluno["nota"]

    if aluno["nota"] > maior_nota:
        maior_nota = aluno["nota"]
        melhor_aluno = aluno["nome"]

media = soma / len(alunos)

print("\nMédia da turma:", round(media,2))
print("Melhor aluno:", melhor_aluno, "-", maior_nota)

File: Java/CadastroAlunos.java
import java.util.ArrayList;
import java.util.Scanner;

class Aluno{
    String nome;
    double nota;

    Aluno(String nome, double nota){
        this.nome = nome;
        this.nota = nota;
    }
}

public class CadastroAlunos {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        ArrayList<Aluno> alunos = new ArrayList<>();

        while(true){

            System.out.print("Nome do aluno: ");
            String nome = sc.nextLine();

            System.out.print("Nota do aluno: ");
            double nota = sc.nextDouble();
            sc.nextLine();

            alunos.add(new Aluno(nome, nota));

            System.out.print("Cadastrar outro aluno? (s/n): ");
            String resp = sc.nextLine();

            if(!resp.equalsIgnoreCase("s")){
                break;
            }
        }

        double soma = 0;
        double maiorNota = 0;
        String melhorAluno = "";

        System.out.println("\nLista de alunos:");

        for(Aluno a : alunos){

            System.out.println(a.nome + " - " + a.nota);

            soma += a.nota;

            if(a.nota > maiorNota){
                maiorNota = a.nota;
                melhorAluno = a.nome;
            }
        }

        double media = soma / alunos.size();

        System.out.println("\nMédia da turma: " + media);
        System.out.println("Melhor aluno: " + melhorAluno + " - " + maiorNota);

        sc.close();
    }
}